### PR TITLE
Technical debt: Pass start script to TS.

### PR DESCRIPTION
**Pull request by Heyrows**
## Description

PR to pass the Start script of slice-machine to typescript in order to prevent unwanted breaking changes.

### CODE REVIEW

File: packages/slice-machine/package.json [hunk 1]
```diff
@@ -35,7 +36,7 @@
     "audit-fix": "npm audit fix"
   },
   "bin": {
-    "start-slicemachine": "start/index.js"
+    "start-slicemachine": "bins/start.js"
```
Comment (MarcMcIntosh) [inline packages/slice-machine/package.json]: traditionally it's called `bin/`

File: packages/slice-machine/server/tsconfig.json [hunk 2]
```diff
@@ -23,5 +24,6 @@
       "@models/*": ["../lib/models/*"]
     }
   },
-  "include": ["./**/*.ts", "./**/*.js", "../lib/**/*.ts", "../lib/**/*.js"]
+  "exclude": ["../build"],
```
Comment (MarcMcIntosh) [inline packages/slice-machine/server/tsconfig.json]: does adding build change anything?

**Review by MarcMcIntosh — COMMENTED** [review]
Interesting thing that could be done is to export the module from `start/index.ys` and call the function in `bin/start.js`

I guess it's out of the scope of this ticket

File: packages/slice-machine/server/tsconfig.json (hunk 2 above)
Comment (Heyrows) [reply packages/slice-machine/server/tsconfig.json]: If a file from the build is imported directly then it avoid to try to transpile it. (we had this for the last start script).

File: packages/slice-machine/package.json (hunk 1 above)
Comment (Heyrows) [reply packages/slice-machine/package.json]: Ok, will change that

**Comment by Heyrows** [comment]
@MarcMcIntosh To answer to your proposition, we wanted to do it that way, but both would have worked 😄

File: packages/core/src/core/communication.ts [hunk 3]
```diff
@@ -37,38 +41,17 @@ export async function refreshSession(
...
-      fold<t.Errors, RepoData, RepoData>(
-        () => {
-          throw new Error("Can't parse repo data");
-        },
-        (f: RepoData) => {
-          return f;
-        }
-      )
-    );
+    return fold<t.Errors, RepositoriesRoles, RepositoriesRoles>(
```
Comment (Miniplop) [inline packages/core/src/core/communication.ts:49]: I like that, when we remove extra complexity for improving the clarity :)

File: packages/slice-machine/server/src/resolveAliases.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import path from "path";
```
Comment (Miniplop) [inline packages/slice-machine/server/src/resolveAliases.ts]: As this file is used by script and node server, we can put this file into the lib folder na ?

File: packages/slice-machine/tests/scripts/common/findArgument.test.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { findArgument } from "../../../scripts/common/findArgument";
```
Comment (Miniplop) [inline packages/slice-machine/tests/scripts/common/findArgument.test.ts:1]: GG

**Review by Miniplop — COMMENTED** [review]
I think you commited too many lock files :p, plus I would expect a little unit test on the resolve alias function.

File: packages/slice-machine/server/src/resolveAliases.ts (hunk 4 above)
Comment (Heyrows) [reply packages/slice-machine/server/src/resolveAliases.ts]: good point, i didn't know where to put it.

File: packages/core/src/core/communication.ts [hunk 6]
```diff
@@ -37,38 +41,17 @@ export async function refreshSession(
...
-);
-export type UserInfo = {
-  userId: string;
-  email: string;
-  type: string;
-  repositories: RepoData;
-};
-
-export function maybeParseRepoData(repos?: string | RepoData): RepoData {
+export function maybeParseRepoData(
```
Comment (MarcMcIntosh) [inline packages/core/src/core/communication.ts:44]: I can't remember where this function was used, but if it's parameters and return type have changed so should the name,

**Review by MarcMcIntosh — APPROVED** [review]

**Pull request merged**